DataFormats/HcalDigi/interface/HcalDigiSoA.h

```diff
@@ -0,0 +1,39 @@
+#ifndef DataFormats_HcalDigi_HcalDigiSoA_h
+#define DataFormats_HcalDigi_HcalDigiSoA_h
+
+//TODO: Use Eigen column for data(?)
+//#include <Eigen/Core>
+//#include <Eigen/Dense>
+
+#include "DataFormats/Common/interface/StdArray.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+
+namespace hcal {
+
+  using QIE11dataArray = edm::StdArray<uint16_t, QIE11DigiCollection::MAXSAMPLES>;
+  using QIE10dataArray = edm::StdArray<uint16_t, HBHEDataFrame::MAXSAMPLES>;
+
+  //using QIE11dataVector = Eigen::Matrix<uint16_t,  QIE11DigiCollection::MAXSAMPLES, 1>;
+  //using QIE10dataVector = Eigen::Matrix<uint16_t,  HBHEDataFrame::MAXSAMPLES, 1>;
+
+  GENERATE_SOA_LAYOUT(HcalPhase1DigiSoALayout,
+                      SOA_COLUMN(uint32_t, ids),
+                      //SOA_EIGEN_COLUMN(QIE11dataVector, data),
+                      SOA_COLUMN(QIE11dataArray, data),
+                      SOA_SCALAR(uint32_t, stride),
```

Comment:
Howe are the `stride` scalars used ?

Reply:
If I'm searching through the code properly, they seem to be set but never used after the SoA has been filled.
What should they represent ?

Reply:
`stride` is used in `Mahi.dev.cc` here
https://github.com/cms-sw/cmssw/pull/44910/files#diff-7a01f10a57585472bcc02bbcddffcdfba8174cafd87f765c03ede07bdf286b51R378-R384
which computes the number of samples for each flavors.

Reply:
OK, thanks for the clarification.